src/Util/ComposerFile.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GrumPHP\Util;
+
+use GrumPHP\Exception\RuntimeException;
+
+class ComposerFile
+{
+    private $configuration;
+
+    public function __construct(array $configuration)
+    {
+        $this->configuration = $configuration;
+    }
+
+    public static function createFrom(string $composerFileLocation): self
+    {
+        if (!\file_exists($composerFileLocation)) {
+            throw new RuntimeException(sprintf('composer.json not found on location %s', $composerFileLocation));
+        }
+        $json = \file_get_contents($composerFileLocation);
+
+        return new self(json_decode($json, true));
+    }
+
+    /**
+     * Composer contains some logic to prepend the current bin dir to the system PATH.
+     * To make sure this application works the same in CLI and Composer mode,
+     * we'll have to ensure that the bin path is always prefixed.
+     *
+     * @see https://github.com/composer/composer/blob/1.1/src/Composer/EventDispatcher/EventDispatcher.php#L147-L160
+     */
+    private static function ensureProjectBinDirInSystemPath(string $binDir): bool
+    {
+        $pathStr = 'PATH';
+        if (!isset($_SERVER[$pathStr]) && isset($_SERVER['Path'])) {
+            $pathStr = 'Path';
+        }
+
+        if (!is_dir($binDir)) {
+            return false;
+        }
+
+        // add the bin dir to the PATH to make local binaries of deps usable in scripts
+        $binDir = realpath($binDir);
+        $hasBindDirInPath = preg_match(
+            '{(^|'.PATH_SEPARATOR.')'.preg_quote($binDir).'($|'.PATH_SEPARATOR.')}',
+            $_SERVER[$pathStr]
+        );
+
+        if (!$hasBindDirInPath && isset($_SERVER[$pathStr])) {
+            $_SERVER[$pathStr] = $binDir.PATH_SEPARATOR.getenv($pathStr);
+            putenv($pathStr.'='.$_SERVER[$pathStr]);
+        }
+
+        return true;
+    }
+
+    public function getBinDir(): string
+    {
+        $binDir = $this->configuration['bin-dir'] ?? 'vendor/bin';
+
+        if (self::ensureProjectBinDirInSystemPath($binDir)) {
+            return $binDir;
+        }
+
+        throw new RuntimeException('Could not get bin dir location');
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getConfigDefaultPath()
+    {
+        return null;
```

Comment:
Shouldn't you check and return following param?
`$extra['grumphp']['config-default-path'];`

Reply:
Ow yes. Clearly something I forgot todo.